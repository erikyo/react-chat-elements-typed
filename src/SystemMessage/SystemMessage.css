.rce-container-smsg {
    display: flex;
    align-items: center;
    justify-content: center;
}

.rce-smsg {
    position: relative;
    background: white;
    display: flex;
    flex-direction: column;
    margin: 5px 0;
    float: left;
    max-width: 70%;
    align-items: center;
    justify-content: center;
    padding: 5px 12px 6px;
    font-size: 12px;
    line-height: 21px;
    color: #54656f;
    text-align: center;
    text-shadow: 0 1px 0 rgb(255 255 255 / 40%);
    border-radius: 8px;
    box-shadow: var(--rce-mbox-shadow);
}

.rce-smsg-text {
    text-align: center;
    display: inline-block;
    font-size: var(--rce-mbox-font-size);
}
