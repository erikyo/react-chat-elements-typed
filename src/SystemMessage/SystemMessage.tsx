import type { FC } from "react";
import "./SystemMessage.css";

import classNames from "classnames";
import type { ISystemMessageProps } from "../types";

const SystemMessage: FC<ISystemMessageProps> = (props) => {
	return (
		<div className={classNames("rce-container-smsg", props.className)}>
			<div className="rce-smsg">
				<div className="rce-smsg-text">{props.text}</div>
			</div>
		</div>
	);
};

export default SystemMessage;
