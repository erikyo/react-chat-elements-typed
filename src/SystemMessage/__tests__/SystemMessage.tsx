import { describe, expect, it } from "vitest";
import React from "react";
import SystemMessage from "../SystemMessage";
import { render } from "@testing-library/react";

describe("Avatar component", () => {
	it("should render without issues", () => {
		const component = render(<SystemMessage />);
		expect(component.container).toBeTruthy();
		expect(component.container).toMatchSnapshot();
	});
});
