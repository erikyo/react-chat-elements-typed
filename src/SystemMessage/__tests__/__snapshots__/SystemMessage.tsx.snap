// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > should render without issues 1`] = `
<div>
  <div
    class="rce-container-smsg"
  >
    <div
      class="rce-smsg"
    >
      <div
        class="rce-smsg-text"
      />
    </div>
  </div>
</div>
`;
